### PR TITLE
CMCL-0000: add null checks for brain camera

### PR DESCRIPTION
**Pull request by glabute**
Missing null checks

### CODE REVIEW

**Review by gaborkb — APPROVED** [review]

**Review by AntoineCharton — APPROVED** [review]

File: com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#if UNITY_EDITOR //Selection.activeGameObject
+using System.Collections;
+using NUnit.Framework;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Unity.Cinemachine.Tests.Editor
```
Comment (glabute) [inline com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs:8]: You're using the Editor namespace in the runtime assembly.  Is this an editor test or a runtime test?

image

File: com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+{
+    [TestFixture]
+    public class CinemachineBrainSetup
+    {
+        CinemachineCamera m_Vcam;
+        CinemachineBrain m_Brain;
+        GameObject m_FollowObject;
```
Comment (glabute) [inline com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs:15]: This is not getting cleaned up

File: com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+using System.Collections;
+using NUnit.Framework;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Unity.Cinemachine.Tests.Editor
+{
+    [TestFixture]
+    public class CinemachineBrainSetup
```
Comment (glabute) [inline com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs:11]: You should be inheriting CinemachineFixtureBase, which handles cleanup of objects.  Look at the other tests for examples.

File: com.unity.cinemachine/Tests/Runtime/MockControllers/InputController1AllAxis.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+using System;
```
Comment (glabute) [inline com.unity.cinemachine/Tests/Runtime/MockControllers/InputController1AllAxis.cs:1]: I think the MockController tests belong in another PR

Comment (AntoineCharton) [reply com.unity.cinemachine/Tests/Runtime/MockControllers/InputController1AllAxis.cs:1]: Ok will remove.

File: com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs (hunk 3 above)
Comment (AntoineCharton) [reply com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs]: Done

File: com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs (hunk 2 above)
Comment (AntoineCharton) [reply com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs:14]: Fixed.

File: com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs (hunk 1 above)
Comment (glabute) [reply com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs:8]: If you look at StateDrivenCameraTests you'll see that it's in the proper runtime namespace, and that it has #if UNITY_EDITOR to ensure that it only runs in the context of editor.  You should do it that way.

Comment (AntoineCharton) [reply com.unity.cinemachine/Tests/Runtime/CinemachineBrainSetupTest.cs:8]: Got it I missed it thanks for catching it.
